test/puppetlabs/puppetdb/query/ast_test.clj

```diff
@@ -0,0 +1,132 @@
+(ns puppetlabs.puppetdb.query.ast-test
+  "Tests the behavior of AST queries."
+  (:require
+   [clj-time.coerce :as coerce-time]
+   [clj-time.core :refer [now]]
+   [clojure.test :refer :all]
+   [puppetlabs.kitchensink.core :refer [dissoc-in]]
+   [puppetlabs.puppetdb.cli.services :as cli-svcs]
+   [puppetlabs.puppetdb.command :refer [enqueue-command]]
+   [puppetlabs.puppetdb.examples :as examples]
+   [puppetlabs.puppetdb.testutils :refer [block-until-results]]
+   [puppetlabs.puppetdb.testutils.cli :refer [get-catalogs get-factsets]]
+   [puppetlabs.puppetdb.testutils.services :as svc-utils :refer [*server*]]
+   [puppetlabs.trapperkeeper.app :refer [get-service]]))
+
+(deftest ast-queries
```

Comment:
I'm not sure, but I keep thinking these tests should belong somewhere else, in that we've already got a place for this. Perhaps http/facts_test.clj?